source/geometries/Next100EnergyPlane.cc

```diff
@@ -260,119 +220,111 @@ namespace nexus {
 
 
     /// Vacuum volume that encapsulates all elements related to PMTs. ///
+    G4double vacuum_front_length = hole_length_front_ + pmt_stand_out_ + optical_pad_thickn_
+                                  + sapphire_window_thickn_ + tpb_thickn_;
     G4Tubs* vacuum_front_solid =
-      new G4Tubs("HOLE_FRONT", 0., hole_diam_front_/2., hole_length_front_/2., 0., twopi);
-    transl_z = hole_length_front_/2. + hole_length_rear_/2 - offset/2.;
-    G4UnionSolid* vacuum_solid =
-      	new G4UnionSolid("HOLE", vacuum_front_solid, hole_rear_solid,
-			 0, G4ThreeVector(0., 0., transl_z));
-    transl_z =
-      hole_length_front_/2. + hole_length_rear_ + hut_hole_length_/2. - offset/2.;
-    vacuum_solid =
-      	new G4UnionSolid("HOLE", vacuum_solid, hole_hut_solid,
-			 0, G4ThreeVector(0., 0., transl_z));
+      new G4Tubs("HOLE_FRONT", 0., hole_diam_front_/2., vacuum_front_length/2., 0., twopi);
+
+    G4Tubs* vacuum_rear_solid =
+      new G4Tubs("HOLE_REAR", 0., hole_diam_rear_/2., (hole_length_rear_+offset)/2., 0., twopi);
 
-    G4Tubs* stand_out_solid =
-      new G4Tubs("STAND_OUT", 0., hole_diam_front_/2.,
-		 (stand_out_length + offset)/2., 0., twopi);
+    G4Tubs* vacuum_hut_solid =
+      new G4Tubs("HOLE_HUT", 0., hut_int_diam_/2., hut_hole_length_/2., 0., twopi);
+
+    G4UnionSolid* vacuum_solid =
+      new G4UnionSolid("EP_HOLE", vacuum_front_solid, vacuum_rear_solid, 0,
+                      G4ThreeVector(0., 0., (vacuum_front_length+hole_length_rear_)/2.));
 
-    transl_z = - hole_length_front_/2. - stand_out_length/2 + offset/2.;
     vacuum_solid =
-      new G4UnionSolid("HOLE", vacuum_solid, stand_out_solid, 0,
-		       G4ThreeVector(0., 0., transl_z));
-    G4LogicalVolume* vacuum_logic =
-      new G4LogicalVolume(vacuum_solid, vacuum, "HOLE");
+      new G4UnionSolid("EP_HOLE", vacuum_solid, vacuum_hut_solid, 0,
+                       G4ThreeVector(0., 0., vacuum_front_length/2.+hole_length_rear_+hut_hole_length_/2.));
 
+    G4LogicalVolume* vacuum_logic = new G4LogicalVolume(vacuum_solid, vacuum, "EP_HOLE");
 
     /// Sapphire window ///
-    G4Tubs* sapphire_window_solid =
-      new G4Tubs("SAPPHIRE_WINDOW", 0., hole_diam_front_/2.,
-		 sapphire_window_thickn_/2., 0., twopi);
+    G4Tubs* sapphire_window_solid = new G4Tubs("SAPPHIRE_WINDOW", 0., hole_diam_front_/2.,
+                                               (sapphire_window_thickn_ + tpb_thickn_)/2., 0., twopi);
```

Comment:
I don't understand why `tpb_thickn_` is added to the `sapphire_window_thickn_` when creating the `sapphire_window_solid` and later adding another `G4Tubs` for the TPB outside this volume (according to line 262).

Reply:
That's because the `tpb_logic` is placed inside the `sapphire_window_logic` instead of been placed inside `vacuum_logic`. I just maintained the convention used before rewiewing this class, but it would be equivalent to place the `tpb_logic` inside the `vacuum_logic` and deleting the `tpb_thickn_` in the definition of `sapphire_window_solid`.